src/Sylius/Component/Order/Model/Order.php

```diff
@@ -650,7 +604,5 @@ public function removeIdentity(IdentityInterface $identity)
             $identity->setOrder(null);
```

Comment:
This should be removed as identity __must__ be assigned to some order, otherwise it's broken.